saleor/core/tests/test_auth_backend_saleor_header.py

```diff
@@ -0,0 +1,197 @@
+import jwt
+import pytest
+from django.contrib.auth.models import Permission
+from freezegun import freeze_time
+from jwt import ExpiredSignatureError, InvalidSignatureError, InvalidTokenError
+
+from ..auth_backend import JSONWebTokenBackend
+from ..jwt import (
+    JWT_ACCESS_TYPE,
+    JWT_ALGORITHM,
+    create_access_token,
+    create_access_token_for_app,
+    create_refresh_token,
+    jwt_encode,
+    jwt_user_payload,
+)
+from ..permissions import get_permissions_from_names
+
+
+def test_use_saleor_header_as_a_first_try(rf, staff_user, customer_user):
+    staff_access_token = create_access_token(staff_user)
+    customer_access_token = create_access_token(customer_user)
+
+    request = rf.request(
+        HTTP_AUTHORIZATION_BEARER=f"{staff_access_token}",
+        HTTP_AUTHORIZATION=f"JWT {customer_access_token}",
+    )
+    backend = JSONWebTokenBackend()
+    user = backend.authenticate(request)
+    assert user == staff_user
```

Comment:
I would like to see a test where `authorization-bearer` is a valid Saleor token, and `authorization` is not (and is either `Basic` or `Bearer` with an unsupported token) to mirror cases where Saleor's API is sitting behind third-party authentication.

Reply:
Added test for this.